docs/source/asr/speech_intent_slot/configs.rst

```diff
@@ -88,6 +88,11 @@ An example config for the model can be:
       subsampling_factor: 4 # must be power of 2
       subsampling_conv_channels: -1 # -1 sets it to d_model
 
+      # Reduction params
+      reduction: null # pooling, subsampling, or null
+      reduction_position: -2 # Encoder block index or -1 for subsampling at the end of encoder
```

Comment:
What does -2 does?
How about using null to disable it?

Reply:
Agree with @VahidooX ; it's not clear what -2 means.

Reply:
- -2 means no reduction is applied.
- -1 means that reduction is applied at the end of encoder (after the output projection layer)
- Integers 0-17 specify the block index after which the subsampling should be applied.

Having null instead for this value also works. I'll change it.

As for checking the bounds, should I add an assert statement to ensure the value is either None or remains between [-1, len(self.layers)]?

Reply:
Yes, that is the bound to get checked.

Reply:
Added a bound check in the new commit